src/neurostatslib/utils.py

```diff
@@ -4,44 +4,263 @@
 from __future__ import annotations
 
 from functools import partial
-from typing import Optional, Union
+from typing import Iterable, List, Optional, Union
```

Comment:
Just a note: we could consider only supporting 3.10 and up, and get the easier type annotations. Since we're still developing right now...